### ISSUE TITLE
Dreambooth on SD1.5/SD2.1-768 - use prompt "an apple" to generate but got random results

### ISSUE THREAD

**Issue by qiaoshun8888** [huggingface/diffusers#1944]
I used dreambooth to fine-tune the models based on SD1.5 and SD2.1-768 with my photos. When I use my token (`xxx man`) in a prompt to generate, it only gives me something similar to my training images (but very ugly) and somehow the entire prompt was ignored.

The weirdest thing is that when I try to use `an apple`, it gives me these random images

<img src="https://user-images.githubusercontent.com/1053081/211133645-6bdb3ca9-9663-4b19-9461-d68d996bebbd.png" width="200"/> | <img src="https://user-images.githubusercontent.com/1053081/211133647-828eb6c0-05b6-4d1f-82e3-9a3e27292c42.png" width="200"/> | <img src="https://user-images.githubusercontent.com/1053081/211133648-c7313c2b-3ffd-449f-8a4a-e31f1c815729.png" width="200"/> | <img src="https://user-images.githubusercontent.com/1053081/211133650-f774bcff-bb4c-4776-b6ac-d4454ce90a0f.png" width="200"/>

**Anyone knows the reason and fix? Thanks!**

I set up the environment today and followed the [installation instructions](https://github.com/huggingface/diffusers/tree/main/examples/dreambooth).

Here is the script I used to fine-tune the model based on SD1.5. SD2.1-768 is slightly different, but I mainly followed the instructions. 
```
#!/bin/bash

export MODEL_NAME="runwayml/stable-diffusion-v1-5"
# export MODEL_NAME="stabilityai/stable-diffusion-2-1"

export INSTANCE_DIR="training"
export CLASS_DIR="classes"
export OUTPUT_DIR="models/model-sd1-5-xxx-man"
export LOGGING_DIR="logs"

PT="xxx"
CLS="man"
STEPS=2500
NUM_CLASS_IMAGES=200

accelerate launch train_dreambooth.py \
  --train_text_encoder \
  --pretrained_model_name_or_path=$MODEL_NAME \
  --instance_data_dir=$INSTANCE_DIR \
  --class_data_dir=$CLASS_DIR \
  --output_dir=$OUTPUT_DIR \
  --with_prior_preservation --prior_loss_weight=1.0 \
  --instance_prompt="a photo of $PT $CLS" \
  --class_prompt="a photo of $CLS" \
  --resolution=512 \
  --mixed_precision="fp16" \
  --train_batch_size=1 \
  --gradient_accumulation_steps=1 \
  --gradient_checkpointing \
  --learning_rate="2e-6" \
  --lr_scheduler="polynomial" \
  --lr_warmup_steps=200 \
  --max_train_steps=$STEPS \
  --num_class_images=$NUM_CLASS_IMAGES \
  --logging_dir=$LOGGING_DIR \
  --use_8bit_adam \
  --enable_xformers_memory_efficient_attention
```

Code for inference
```
pipe = StableDiffusionPipeline.from_pretrained(
  "models/model-sd1-5-xxx-man",
  revision="fp16",
  torch_dtype=torch.float16,
  safety_checker=None,
).to("cuda")

pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config)
pipe.enable_xformers_memory_efficient_attention()

image = pipe(
  prompt="an apple",
  width=512,
  height=512,
  num_inference_steps=25,
  guidance_scale=7.5,
  negative_prompt='',
  generator=None,
  num_images_per_prompt=1,
).images[0]
...
```

Results of running `diffusers-cli env`
```
- `diffusers` version: 0.12.0.dev0
- Platform: Linux-4.19.0-22-cloud-amd64-x86_64-with-debian-10.13
- Python version: 3.7.12
- PyTorch version (GPU?): 1.13.1+cu116 (True)
- Huggingface_hub version: 0.11.1
- Transformers version: 0.15.0
- Accelerate version: not installed
- xFormers version: 0.0.15+bad4138.d20230106
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>
```

It says Accelerate not installed, but I did install it properly.
```
john@dreambooth-john-v2:~/github/diffusers/examples/dreambooth$ pip list | grep accelerate
accelerate                             0.15.0
john@dreambooth-john-v2:~/github/diffusers/examples/dreambooth$ 
```

**Comment by qiaoshun8888** [comment]
**Updates**

I found that if you train a model with the flag `--use_8bit_adam`, the model will have the issue I mentioned above. 
I have tested it with different configurations. It happens when using `runwayml/stable-diffusion-v1-5` and `stabilityai/stable-diffusion-2-1`, but somehow the flag works on `stabilityai/stable-diffusion-2-1-base`.

Not sure if it's a bug or not. If someone can confirm or test it that will be great!

Thanks.

**Comment by pcuenca** [comment]
Hi @patil-suraj, do you have any insight about that?

**Comment by mingqizhang** [comment]
Maybe you can train without xformer?

**Comment by qiaoshun8888** [comment]
> Maybe you can train without xformer?

I did. I trained multiple models by removing each of the flags that I can remove. Only `--use_8bit_adam` affects the result.

**Comment by mingqizhang** [comment]
The flag --use_8bit_adam works for me, but when i train with xformer, the inference result is not correct, which is the same as pretrained weight runwayml/stable-diffusion-v1-5. Do you know the reason?

**Comment by qiaoshun8888** [comment]
> The flag --use_8bit_adam works for me, but when i train with xformer, the inference result is not correct, which is the same as pretrained weight runwayml/stable-diffusion-v1-5. Do you know the reason?

Do you mean the fine-tuned model didn't learn the object you trained? like totally ignored the token? One of my friends had this issue, but I couldn't reproduce it.

I don't think I had any issues when using xformerx. How did you install it?
I followed the instruction [here](https://github.com/facebookresearch/xformers#installing-xformers) and used this command
`pip install -v -U git+https://github.com/facebookresearch/xformers.git@main#egg=xformers`

**Comment by patrickvonplaten** [comment]
cc @patil-suraj can you check here?

**Comment by patil-suraj** [comment]
I have used `xformers` extensively for training and never had any issue with it, so think it's unlikely that its a `xformers` related issue. The 8-bit-Adam is also used a lot for dreambooth so not really sure if it could be the problem. 


>> I used dreambooth to fine-tune the models based on SD1.5 and SD2.1-768 with my photos. When I use my token (xxx man) in a prompt to generate, it only gives me something similar to my training images (but very ugly) and somehow the entire prompt was ignored.

This smells like overfitting to me, either you are doing too many steps, using a high lr or using less images. Check the blog we wrote which has some tips and tricks to avoid such issues https://huggingface.co/blog/dreambooth
